### PR TITLE
Botirjon/GetOrdersByFilter

### PR DESCRIPTION
**Pull request by Botirjon29**
add-GetOrderByOrganizationId  bolla tekshirovoradigan joyi 👀

### CODE REVIEW

File: FurnitureShop/src/FurnitureShop.Admin.Api/Services/OrdersService.cs [hunk 1]
```diff
@@ -30,4 +30,17 @@ public async Task<OrderView> GetOrderByIdAsync(Guid orderId)
...
+        var organization = _unitOfWork.Organizations.GetById(organizationId);
+        if (organization is null)
+            throw new NotFoundException<Organization>();
+
+        var orders = _unitOfWork.Orders.GetAll().Where(o => o.OrganizationId == organizationId);
+        if (orders is null)
+            throw new NotFoundException<Order>();
+
+        return Task.FromResult(orders.Adapt<List<OrderView>>());
+    }
```
Comment (leverow) [inline FurnitureShop/src/FurnitureShop.Admin.Api/Services/OrdersService.cs:45]: async qilib ko'ringchi

**Review by leverow — CHANGES_REQUESTED** [review]

File: FurnitureShop/src/FurnitureShop.Admin.Api/Services/OrdersService.cs (hunk 1 above)
Comment (javlonfattoev) [inline FurnitureShop/src/FurnitureShop.Admin.Api/Services/OrdersService.cs:45]:
```C#
Task<List<OrderView>> GetOrdersAsync();
```
orgid bilan filter qilishni shu methodga qoshing

```C#
Task<List<OrderView>> GetOrdersAsync(OrderFilter filter);
...
public class OrderFilter : PagenationParams
{
 public Guid? OrganizationId { get; set; }
}
```
orgid kiritsa filtr qilinadi kiritmasa hammasini beradi
shunda bu kerak emas

Comment (javlonfattoev) [reply FurnitureShop/src/FurnitureShop.Admin.Api/Services/OrdersService.cs:45]: async await qilmabsiz

**Review by javlonfattoev — CHANGES_REQUESTED** [review]

**Review by leverow — APPROVED** [review]
Looks good to me 👍

**Comment by javlonfattoev** [comment]
List soralganda notfound qaytarilmaydi bub bush list keladi

**Comment by javlonfattoev** [comment]
var query = _unitOfWork.Orders.GetAll();
if(filter.OrganizationId is not null)
   query = query.Where(o => o.OrganizationId == filter.OrganizationId);

//masalan boshqa parametr uchun
if(filter.UserId is not null)
   query = query.Where(o => o.UserId== filter.UserId);

var orders = query.ToListAsync();
return orders.Adapt<List<OrderView>>();

**Comment by javlonfattoev** [comment]
```C#
var query = _unitOfWork.Orders.GetAll();
if(filter.OrganizationId is not null)
   query = query.Where(o => o.OrganizationId == filter.OrganizationId);

//masalan boshqa parametr uchun
if(filter.UserId is not null)
   query = query.Where(o => o.UserId== filter.UserId);

var orders = query.ToListAsync();
return orders.Adapt<List<OrderView>>();
```

**Review by javlonfattoev — APPROVED** [review]

**Pull request merged**